main_test.go

```diff
@@ -86,6 +86,7 @@ func TestScript(t *testing.T) {
 
 				"gofullversion="+runtime.Version(),
 				"EXEC_PATH="+execPath,
+				"GARBLE_CACHE_DIR="+filepath.Join(filepath.Dir(env.WorkDir), "garble-cache-tmp"),
```

Comment:
what is this for? if it's to share the linker binary between test scripts, I don't think it does that; your cache dir is under WorkDir, which is a new and clean directory for each test script run.

we allow the tests to use the host's GOCACHE by default, by passing the env var along. this happens via `gotooltest.Setup`. you could do the same by passing GARBLE_CACHE_DIR to be the host's UserCacheDir.